docs/standard/analyzers/platform-compat-analyzer.md

```diff
@@ -0,0 +1,366 @@
+      // even there were 3 diagnostics found analyzer warn only for the first 2.
+      StartedWindowsSupportFrom8UnsupportedFrom10();
+  }
+  ```
+
+## Handle reported warnings
+
+The recommended way to deal with these diagnostics is to make sure you only call platform-specific APIs when running on an appropriate platform. Following are the options you can use to address the warnings; choose whichever is most appropriate for your situation:
+
+- **Guard the call**. You can either achieve this by excluding the code at build time using `#if` and multi-targeting or by conditionally calling the code at run time. Check whether you’re running on a desired `Platform` by using one of platform-check methods, for example, `OperatingSystem.Is<Platform>()` or `OperatingSystem.Is<Platform>VersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0)`.
+
+- **Mark the call site as platform-specific**. You can also choose to mark your own APIs as being platform-specific, thus effectively just forwarding the requirements to your callers. Mark the containing method or type or the entire assembly with the same attributes as the referenced platform-dependent call. [Examples](#mark-call-site-as-platform-specific).
+
+- **Assert the call site with platform check**. If you don't want the overhead of an additional `if` statement at run time, use <xref:System.Diagnostics.Debug.Assert(System.Boolean)?displayProperty=nameWithType>. [Example](#assert-the-call-site-with-platform-check).
+
+- **Delete the code**. Generally not what you want because it means you lose fidelity when your code is used by Windows users. For cases where a cross-platform alternative exists, you’re likely better off using that over platform-specific APIs.
+
+- **Suppress the warning**. You can also simply suppress the warning, either via editor.config or `#pragma warning disable`. However, this option should be a last resort when using platform-specific APIs.
+
+### Guard platform-specific APIs with guard methods
+
+The guard method's platform name should match with the calling platform-dependent API platform name. If the platform string of the calling API includes the version:
+
+- For the `[SupportedOSPlatform("platformVersion")]` attribute, the guard method platform `version` should be greater than or equal to the calling platform's `Version`.
+- For the `[UnsupportedOSPlatform("platformVersion")]` attribute, the guard method's platform `version` should be less than or equal to the calling platform's `Version`.
+
+  ```csharp
+  public void CallingSupportedOnlyApis() // Allow list calls
+  {
+      if (OperatingSystem.IsWindows())
+      {
+          WindowsOnlyApi(); // not warn
+      }
+
+      if (OperatingSystem.IsLinux())
+      {
+          SupportedOnWindowsAndLinuxOnly(); // nor warn, within one of the supported context
+      }
+
+      if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) // traditional method
```

Comment:
"traditional" is a weird term. We should downplay this API but have a section for if you need to guard code that targets netstandard or netcoreapp as well as net5.0.  In those cases, this API is useful and will be respected. But it's not as optimized.